acdc/TLACDCCorrespondence.py

```diff
@@ -6,17 +6,19 @@
     EdgeType,
 )  # these introduce several important classes !!!
 from acdc.acdc_utils import OrderedDefaultdict, make_nd_dict
-from typing import List, Dict, Optional, Tuple, Union, Set, Callable, TypeVar, Iterable, Any
+from typing import List, Dict, MutableMapping, Optional, Tuple, Union, Set, Callable, TypeVar, Iterable, Any
 
 
 class TLACDCCorrespondence:
     """Stores the full computational graph, similar to ACDCCorrespondence from the rust_circuit code
     
     The two attributes, self.graph and self.edges allow for efficiently looking up the nodes and edges in the graph: see `notebooks/editing_edges.py`"""
-        
+
+    graph: MutableMapping[str, MutableMapping[TorchIndex, TLACDCInterpNode]]
+    edges: MutableMapping[str, MutableMapping[TorchIndex, MutableMapping[str, MutableMapping[TorchIndex, Edge]]]]
     def __init__(self):
-        self.graph: OrderedDict[str, OrderedDict[TorchIndex, TLACDCInterpNode]] = OrderedDefaultdict(OrderedDict) # TODO rename "nodes?"
-        self.edges: OrderedDict[str, OrderedDict[TorchIndex, OrderedDict[str, OrderedDict[TorchIndex, Optional[Edge]]]]] = make_nd_dict(end_type=None, n=4)
+        self.graph = OrderedDefaultdict(OrderedDict) # TODO rename "nodes?"
+        self.edges = make_nd_dict(end_type=None, n=4)
```

Comment:
```suggestion
        self.graph: graph = OrderedDefaultdict(OrderedDict) # TODO rename "nodes?"
        self.edges: edges = make_nd_dict(end_type=None, n=4)
```

Unless I'm missing something? 
I think we should probably have some more type safety here. 
Would you be happy with me making a follow up PR that checks that the strings used to index into `self.graph` and `self.edges` are actually names of HookPoints? Currently, defaultdict is super permissive here

Reply:
1. `graph` and `edges` are not types
2. the type of these attributes is set at the top-level of the class. This also shows that they're the class's only attributes

Yeah, I'd be happy with this PR. I think the right way to do it is to add a `commit()` method that changes the defaultdicts to dicts, and call it after `setup_from_model`. Or, even safer but more annoying maybe, make `add_node` and `add_edge` create the new dicts manually.